### PR TITLE
got all tests passing for js and changed order of test gen templates

### PR DESCRIPTION
**Pull request by parrt**

### CODE REVIEW

File: bild.py [hunk 1]
```diff
@@ -68,11 +68,7 @@
 	"CSharp"   : uniformpath(CSHARP_TARGET)+"/tool/test/org/antlr/v4/test/runtime/csharp/CSharp.test.stg",
 	"Python2"  : uniformpath(PYTHON2_TARGET)+"/tool/test/org/antlr/v4/test/runtime/python2/Python2.test.stg",
 	"Python3"  : uniformpath(PYTHON3_TARGET)+"/tool/test/org/antlr/v4/test/runtime/python3/Python3.test.stg",
-	# "NodeJS"   : uniformpath(JAVASCRIPT_TARGET)+"/tool/test/org/antlr/v4/test/rt/js/node/NodeJS.test.stg",
-	# "Safari"   : uniformpath(JAVASCRIPT_TARGET)+"/tool/test/org/antlr/v4/test/rt/js/safari/Safari.test.stg",
```
Comment (sharwell) [inline bild.py]: :question: Are there plans to include these tests later?

Comment (parrt) [inline bild.py]: will ask Eric

File: tool/test/org/antlr/v4/test/runtime/java/Java.test.stg [hunk 2]
```diff
@@ -148,6 +110,44 @@ public void test<test.name>() throws Exception {
 
 >>
 
+testAnnotations(test) ::= <%
```
Comment (sharwell) [inline tool/test/org/antlr/v4/test/runtime/java/Java.test.stg]: :question: is this a straight copy-paste or should I be looking at content as well?

File: tool/tool.iml [hunk 3]
```diff
@@ -34,6 +35,15 @@
     <content url="file://$MODULE_DIR$/../../antlr4-csharp/tool/test">
       <sourceFolder url="file://$MODULE_DIR$/../../antlr4-csharp/tool/test" isTestSource="false" />
     </content>
+    <content url="file://$MODULE_DIR$/../../antlr4-javascript">
+      <sourceFolder url="file://$MODULE_DIR$/../../antlr4-javascript/tool/src" isTestSource="false" />
+      <sourceFolder url="file://$MODULE_DIR$/../../antlr4-javascript/tool/test" isTestSource="false" />
+      <excludeFolder url="file://$MODULE_DIR$/../../antlr4-javascript/tool/test/org/antlr/v4/test/rt" />
+      <excludeFolder url="file://$MODULE_DIR$/../../antlr4-javascript/tool/test/org/antlr/v4/test/runtime/javascript/chrome" />
```
Comment (sharwell) [inline tool/tool.iml]: :question: Are these excludes needed (the browsers, from `chrome` to `safari` below)? It was my understanding that generation for these folders is currently disabled.

Comment (parrt) [inline tool/tool.iml]: just me getting ide to build. can ignore.

**Comment by sharwell** [comment]
:+1: Changes look fine. I assumed the changes to **Java.test.stg** were just cut/paste existing templates in the file. I would need to look again if the content actually changed.

File: tool/test/org/antlr/v4/test/runtime/java/Java.test.stg (hunk 2 above)
Comment (parrt) [inline tool/test/org/antlr/v4/test/runtime/java/Java.test.stg]: should be copy

**Comment by parrt** [comment]
just reordered. no content change.

**Pull request merged**